### PR TITLE
V4 plugin

### PR DESCRIPTION
**Pull request by ulrikpedersen**
This PR contain the additions for the EPICS V4 plugin/driver by @brunoseivam - to be merged into master. This is the result of the work outlined in ticket #135 

This is a fairly extensive addition to ADCore. Before accepting this PR to merge into master, could everybody (at least @ajgdls, @MarkRivers, @dhickin and @brunoseivam) carefully review, test and verify that the list of requested features have been met?

In summary from #135 :

- [x] Merge master back into the v4-plugin branch, 
 - [x] make sure it still works.
- [x] Make a v4-plugin branch in areaDetector/areaDetector that adds what is need to .gitmodules, EXAMPLE_RELEASE_PATHS.local, EXAMPLE_CONFIG_SITE.local, Makefile, INSTALL.md, documentation/Doxyfile, etc.
- [x] Make a v4-plugin branch in ADExample that allows testing the plugin and driver with simDetector.
- [x] It is easy to disable building the V4 plugin/driver
- [ ] When building is enabled it works without problems on all platforms
- [x] There is (HTML) documentation like all other plugins and drivers
- [ ] There are operator displays for all display managers to control them
- [x] They can be tested in the example simDetector IOC in ADExample

### CODE REVIEW

**Comment by brunoseivam** [comment]
I opened pull requests for [areaDetector](https://github.com/areaDetector/areaDetector/pull/17) and [ADExample](https://github.com/areaDetector/ADExample/pull/1)

**Comment by ulrikpedersen** [comment]
Thanks @brunoseivam, I have ticked off the relevant tasks from this PR. The details of the implementation can be reviewed through each of those individual PRs.

**Comment by ulrikpedersen** [comment]
Reviewing the list of tickboxes:

> It is easy to disable building the V4 plugin/driver

:+1: Done. Just set the build variable EPICS_V4=YES|NO. Added to the EXAMPLE_CONFIG_SITE.local in the PR: areaDetector/areaDetector#17

> When building is enabled it works without problems on all platforms

Built and tested successfully on Linux and Windows (at DLS). Will anyone test on VxWorks? @MarkRivers?

> There is (HTML) documentation like all other plugins and drivers

:+1: Done. Missing a link to pvaDriverDoc.html from the main areaDetectorDoc.html page.

> There are operator displays for all display managers to control them

There are EDM and CSS screens. Will anyone add MEDM screens? There are only very few PVs...

> They can be tested in the example simDetector IOC in ADExample

:+1: There is a "pvaDriverIOC" added in PR: areaDetector/ADExample#1

In summary: we have been doing extensive testing at Diamond - and added some fixes which are included in this PR. I am happy with the state of this plugin and driver and would like to see it merged into master shortly.

**Comment by ulrikpedersen** [comment]
Hi @MarkRivers 

I notice that you've been updating the areaDetector/ADCore v4-plugin branch with the latest from master... Please be aware that that branch is out of date - this PR comes from Brunos branch at brunoseivam/ADCore@v4-plugin. 

If you are testing then you should fetch the "v4-plugin" branch from Brunos fork. See for example the "Commandline instructions" at the bottom of this PR.

Cheers,
Ulrik

**Comment by MarkRivers** [comment]
Can we change this pull request to pull into areaDetector/ADCore/v4-plugin rather than into master?  Then have a new pull request from areaDetector/ADCore/v4-plugin  into areaDetector/ADCore/master.  That way I can make fixes to the v4-plugin branch if I find problems with Windows before merging into master.

**Comment by ulrikpedersen** [comment]
> Can we change this pull request to pull into areaDetector/ADCore/v4-plugin rather than into master? 

I'm afraid it's not possible to "redirect" a PR...

I have now just updated the areaDetector/ADCore v4-plugin branch with Brunos branch. I suggest that you fetch+merge (or pull) the v4-plugin branch again and then build+test locally on that branch. If you do find that changes are required, you then have 3 options:

* Raise a Pull Request back to Brunos v4-plugin branch (before we merge it into areaDetector/ADCore master). It will then automatically become part of the history of the original PR.
* Or merge Brunos v4-plugin branch into areaDetector/ADCore master followed by a new Pull Request from your areaDetector/ADCore v4-plugin branch.
* Discard this PR (close without merging) and raise a new one from areaDetector/ADCore v4-plugin branch.

We have built and tested this on Windows at DLS. The V4 libraries were built by @dhickin who may have some insight to any tweaks required for windows...

**Comment by MarkRivers** [comment]
I am using Ulrik's option 3, closing this pull request, and will raise a new one from areaDetector/ADCore v4-plugin branch.
